src/main/java/com/gregtechceu/gtceu/api/capability/IThermalFluidHandlerItemStack.java

```diff
@@ -26,12 +24,12 @@ public interface IThermalFluidHandlerItemStack {
     default boolean canFillFluidType(FluidStack stack) {
         if (stack == null || stack.getFluid() == null) return false;
 
-        Fluid fluid = stack.getFluid();
-        var temp = FluidHelper.getTemperature(stack);
+        FluidType fluidType = stack.getFluid().getFluidType();
+        var temp = fluidType.getTemperature();
         if (temp > getMaxFluidTemperature()) return false;
         // fluids less than 120K are cryogenic
         if (temp < 120 && !isCryoProof()) return false;
-        if (FluidHelper.isLighterThanAir(stack) && !isGasProof()) return false;
+        if (fluidType.isLighterThanAir() && !isGasProof()) return false;
```

Comment:
`FluidType#getLighterThanAir` does not have a stack-aware version. 
Internally, this is Forge's implementation of the method: `return this.getDensity() <= 0;` 
In fact, the original LDLib FluidHelper method doesn't use a stack-aware version. 
This can be remedied if needed by doing the stack-aware version of the density and comparing it to 0.
I have changed it for any other property getters if applicable.